Library/Homebrew/cmd/vendor-install.sh

```diff
@@ -14,16 +14,15 @@ VENDOR_DIR="${HOMEBREW_LIBRARY}/Homebrew/vendor"
 # Built from https://github.com/Homebrew/homebrew-portable-ruby.
 if [[ -n "${HOMEBREW_MACOS}" ]]
 then
-  if [[ "${HOMEBREW_PROCESSOR}" == "Intel" ]] ||
+  if [[ "$(sysctl -n machdep.cpu.brand_string)" != "Apple"* ]] ||
      # Handle the case where /usr/local/bin/brew is run under arm64.
      # It's a x86_64 installation there (we refuse to install arm64 binaries) so
      # use a x86_64 Portable Ruby.
      [[ "${HOMEBREW_PROCESSOR}" == "arm64" && "${HOMEBREW_PREFIX}" == "/usr/local" ]]
   then
     ruby_FILENAME="portable-ruby-2.6.8.yosemite.bottle.tar.gz"
     ruby_SHA="0cb1cc7af109437fe0e020c9f3b7b95c3c709b140bde9f991ad2c1433496dd42"
-  elif [[ "${HOMEBREW_PROCESSOR}" == "arm64" ]]
-  then
+  else
```

Comment:
We no longer need a check here since the only time we reach this branch is if `sysctl -n machdep.cpu.brand_string` is `Apple*` AND we're not in a `/usr/local` installation of Homebrew.

Additionally, keeping the original check will mean that a Portable Ruby will not be installed for native ARM installations of Homebrew running under Rosetta.